feat: add git hooks and lint staged [gh-516]

nahuelon opened the pull request:
I hereby confirm that I followed the code guidelines found at [engineering guidelines](https://www.notion.so/checkly/Engineering-Guidelines-a3a165a203a04dc1a112f0e26b2f2d3f)

## Affected Components
* [x] CLI
* [x] Create CLI
* [x] Test
* [x] Docs
* [x] Examples
* [x] Other

<!-- You can erase any parts of this template not applicable to your Pull Request. -->
## Notes for the Reviewer
This PR modifies `cli` and `create-cli` packages adding `lint-staged`, `commitlint`, and `simple-git-hooks` hooks packages in the monorepo `package.json`. Some adjustments were applied to follow a monorepo project structure.

1. remove `examples/` from the workspaces lists
2. remove duplicated configuration for `eslint` (`package.json` section)
3. centralize linting and hooks by moving `eslint` and `simple-git-hooks` related packages from each workspace to the monorepo project
4. install the `simple-git-hooks` and `lint-staged` packages as `devDependencies` (and `prepare` script) within the monorepo project
5. adjust `CONTRIBUTING.md` docs.
6. fix linting small issues for `fixtures` that can compliance the linting rules, and disable `eslint` for files that are used to test 'syntax-errors'. Also, the are some fixtures files with some eslint rules disabled (like console.log, required, no-new
7. fix linting for `/examples` folder
8. adjust Github Actions to use monorepo project
<!-- Anything the reviewer should pay extra attention to. -->

Lot of files were modified (small changes) because I had to compliance all the linting rules before committing. This results in a very cleaned up source code.

All this changes are part of https://github.com/checkly/checkly-cli/issues/516 tasks.
## New Dependency Submission
<!-- Please explain here why we need the new dependency. -->

umutuzgur commented on packages/cli/constructs.js:
```diff
@@ -1 +1,3 @@
-module.exports = require('./dist').constructs
+import { constructs } from './dist'
```
I think this has to stay the way it was, or?

nahuelon replied:
I think we can use the `import`, but I'm trying it locally with different type of projects.

umutuzgur replied:
Because this is a module type

nahuelon replied:
I'll revert this change for now.

umutuzgur commented on commitlint.config.js:
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  extends: ['@commitlint/config-conventional'],
```
Gotcha, my mistake

nahuelon commented:
@umutuzgur , I adjusted all the previous linting configuration to use a monorepo approach and take advantage of its benefits. Now, dependency packages were removed from each project (`cli` and `create-cli`) and added to the monorepo project. I updated the PR description adding all changes the PR includes.

umutuzgur commented on packages/create-cli/src/actions/template.ts:
```diff
@@ -22,7 +22,7 @@ export async function copyTemplate ({ template, templatePath, targetDir }: CopyT
       downloadTemplateSpinner.fail()
     } else {
-      console.error(e.message)
+      process.stderr.write(e.message)
```
There is a difference between console and stderr.write. console operations always add a new line whereas stderr.write doesn't

umutuzgur reviewed (approved):
LGTM
